### PR TITLE
[bitnami/opensearch] Fix plugins installation

### PR DESCRIPTION
**Pull request by juan131**
### Description of the change

This PR fixes the conditions under init containers are rendered in OpenSearch pods and it fixes custom plugins installation by ensuring we respect original container permissions on `plugins` and `config` directories via the "copy-default-plugins" init container.

### Benefits

User can install custom plugins without issues.

### Possible drawbacks

None

### Applicable issues

- fixes https://github.com/bitnami/charts/issues/29498

### Additional information

N/A

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/opensearch/templates/master/statefulset.yaml [hunk 1]
```diff
@@ -107,20 +106,16 @@ spec:
           resources: {{- include "common.resources.preset" (dict "type" .Values.volumePermissions.resourcesPreset) | nindent 12 }}
           {{- end }}
           volumeMounts:
-            - name: empty-dir
-              mountPath: /tmp
-              subPath: tmp-dir
```
Comment (andresbono) [inline bitnami/opensearch/templates/master/statefulset.yaml:112]: Why are you removing this?

**Review by andresbono — COMMENTED** [review]
Are these changes needed for bitnami/elasticsearch as well?

File: bitnami/opensearch/templates/master/statefulset.yaml (hunk 1 above)
Comment (juan131) [reply bitnami/opensearch/templates/master/statefulset.yaml:112]: The init-container isn't using the `/tmp` folder

**Comment by juan131** [comment]
> Are these changes needed for bitnami/elasticsearch as well?

They'll be done in a separate PR

**Comment by andresbono** [comment]
Please, check the CI result, something is wrong with the config directory

**Review by andresbono — APPROVED** [review]

**Pull request merged**